src/master/udp_rpc.hpp

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (C) 2008 Search Solution Corporation. All rights reserved by Search Solution.
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ */
+
+/*
+ * udp_rpc.hpp - remote procedure call (RPC) over UDP transport module
+ */
+
+#ifndef _UDP_RPC_HPP_
+#define _UDP_RPC_HPP_
+
+#include "hostname.hpp"
+
+#include <functional>
+#include <map>
+#include <thread>
+
+namespace cubhb
+{
+
+  // type aliases
+  using socket_type = SOCKET;
+  using ipv4_type = std::uint32_t;
+  using port_type = std::uint16_t;
+
+  // constants
+  static const std::size_t BUFFER_SIZE = 4096;
+
+  enum message_type
+  {
+    UNKNOWN_MSG = 0,
+    HEARTBEAT = 1
+  };
+
+  class body_type
```

Comment:
_type  suffix suggests a meta-data on body, but in fact it looks more like it comprises a an encoding of Body
What about another name ?: payload, message, ...